modules/sdk/src/main/scala/org/tessellation/sdk/http/p2p/clients/L0SnapshotClient.scala

```diff
@@ -4,35 +4,68 @@ import cats.effect.Async
 
 import org.tessellation.kryo.KryoSerializer
 import org.tessellation.schema.SnapshotOrdinal
-import org.tessellation.schema.snapshot.Snapshot
+import org.tessellation.schema.snapshot.{Snapshot, SnapshotInfo, SnapshotMetadata}
+import org.tessellation.sdk.domain.cluster.services.Session
 import org.tessellation.sdk.domain.http.p2p.SnapshotClient
 import org.tessellation.sdk.http.p2p.PeerResponse
 import org.tessellation.sdk.http.p2p.PeerResponse.PeerResponse
 import org.tessellation.security.SecurityProvider
+import org.tessellation.security.hash.Hash
 import org.tessellation.security.signature.Signed
 
 import io.circe.Decoder
 import io.circe.magnolia.derivation.decoder.semiauto._
 import io.circe.refined._
+import org.http4s.Method.GET
 import org.http4s.client.Client
 
-object L0SnapshotClient {
+abstract class L0SnapshotClient[
```

Comment:
if this is an abstract class, why you specify a separate trait for definitions?
imo either you can use that abstract class as a type, or move the implementation to trait (which is also possible). I don't know which option would be better, I think it's up to you. what do you think?

Also, I think that there should be:
```
SnapshotClient - this is that class, the base one
GlobalSnapshotClient - (initialized on the Global L0 layer with session and on the other layers with no session)
CurrencySnapshotClient - (initialized on the Currency L0 layer with session and on the other layers with no session)
```
`L0SnapshotClient` name can be misleading as it's a generic implementation now

Reply:
Yes, I agree we can remove the trait and rename the abstract class.

I'm confused about your second comment: The implementation for the `CurrencySnapshotClient` was refactored to use the session. So should it do that, or not? Or should there be another client for currency that does not use the session?

Unless I'm missing something there are only 2 clients in use in the codebase right now:
The global client with no session
The currency client with the session

Reply:
Changes pushed

Reply:
as written on slack yesterday, there should be four (or three) clients:
- The global client with session (should be used in Global L0 download)
- The global client with no session (should be used in DAG L1 and Currency L0/L1)
- The currency client with session (should be used on Currency L0)
- The currency client with no session (should be used on Currency L1 if needed. probably not needed)

Reply:
@marcinwadon Ok, I think the safest thing to do is remove the global snapshot client from `SdkP2PClient` otherwise it's confusing/error-prone how to create it and too easy to use the wrong client in the wrong place. Then anyone who needs to create the Global and Currency Snapshot Clients should explicitly create the one they need.

Reply:
Good idea :+1: as long as it's not used in the SDK module (but definitions imo should be in the sdk)

Reply:
Ok, changes pushed. I removed the global client from the there and now the `P2PClient` objects in each module define the global (and currency, if needed) snapshot clients appropriately for that module.